File: python/leetcode/226. Invert Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if root is not None and (root.right is not None
                                 or root.left is not None):
            tmp = root.right
            root.right = root.left
            root.left = tmp

            self.invertTree(root.right)
            self.invertTree(root.left)

        return root

File: go/leetcode/226. Invert Binary Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
    if root != nil && (root.Right != nil || root.Left != nil) {
        tmp := root.Right
        root.Right = root.Left
        root.Left = tmp
        
        invertTree(root.Left)
        invertTree(root.Right)
    }
    
    return root    
}